### ISSUE TITLE
roslyn for benchmark.

### ISSUE THREAD

**Issue by NMSAzulX** [dotnet/roslyn#35782]
Who can tell me how to test the performance of roslyn?
This is the source issue [address](https://github.com/dotnet/BenchmarkDotNet/issues/1152).

**Comment by jcouv** [comment]
I don't know much about it, but we have compiler benchmarks at https://github.com/dotnet/roslyn/tree/master/src/Tools/CompilerBenchmarks
Does that address your question?

**Comment by NMSAzulX** [comment]
> I don't know much about it, but we have compiler benchmarks at https://github.com/dotnet/roslyn/tree/master/src/Tools/CompilerBenchmarks

**Thanks .I tried the example you gave me, and it still doesn't work. I don't know how it works. I'm just piecing it together.**  
  
----------------------------------------

> Does that address your question?    


**Yes. That's my problem address, and source code.** 

Because the process of dynamic compilation need loading assemblies, and I use "<PreserveCompilationContext>true</PreserveCompilationContext>" . The above may be different from the normal runtime environment. I don't think benchmark's runtime environment supports dynamic compilation very well. 

I saw the following error in the console:
 "**Wrong assembly binding redirects for Microsoft.CodeAnalysis.CSharp.resources, Version=3.0.0.0, Culture=zh-CN, PublicKeyToken=31bf3856ad364e35. 
The value of RuntimeMetadataVersion could not be found. The assembly containing System.Object could not be found, or a value was not specified for RuntimeMetadataVersion with the option.** "
